packages/components/src/unit-control/index.tsx

```diff
@@ -168,6 +171,50 @@ function UnforwardedUnitControl(
 				refInputSuffix.current?.focus();
 			}
 		};
+		onPaste = ( event: ClipboardEvent< HTMLInputElement > ) => {
+			props.onPaste?.( event );
+			const pastedText = event.clipboardData.getData( 'text' );
+			const parsed = parseQuantityAndUnitFromRawValue(
+				pastedText,
+				units
+			);
+			// If no unit parsed, returns early to let the default paste happen.
+			if ( ! parsed[ 1 ] ) return;
+			const [ quantityPasted, unitPasted ] = parsed;
+			event.preventDefault();
+			const changeProps = {
+				event,
+				data: units.find( ( item ) => item.value === unitPasted ),
+			};
+
+			if ( quantityPasted ) {
+				const entry = `${ quantityPasted }`.replace( /^0*/, '' );
+				let {
+					value,
+					selectionStart: headIndex,
+					selectionEnd: tailIndex,
+				} = event.target as HTMLInputElement;
+				headIndex ??= 0;
+				tailIndex ??= 0;
+				if ( headIndex !== tailIndex ) {
+					// Replaces selection with the pasted quantity.
+					const selected = value.substring( headIndex, tailIndex );
+					value = value.replace( selected, `${ entry }` );
+				} else {
+					// Inserts the pasted quantity at caret.
+					const head = value.substring( 0, headIndex );
+					const tail = value.substring( headIndex, value.length );
+					value = `${ head }${ entry }${ tail }`;
+				}
+				onChangeProp?.( `${ value }${ unitPasted }`, changeProps );
+			}
+			// Moves focus to the UnitSelectControl.
+			refInputSuffix.current?.focus();
+
+			if ( unitPasted === parsedUnit ) return;
+			onUnitChange?.( unitPasted, changeProps );
+			setUnit( unitPasted );
```

Comment:
Is it possible to do the `onChangeProp`/`onUnitChange`/`setUnit` calls via the designated handlers (`handleOnQuantityChange`/`handleOnUnitChange`) or do they need to be called "à la carte" like this?